src/hu_stuff.c

```diff
@@ -387,6 +387,25 @@ void HU_ResetMessageColors(void)
     }
 }
 
+static char* ColorByHealth(mobj_t *mobj)
+{
+  int health = (mobj->player ? mobj->player->health : mobj->health) * 100 / mobj->info->spawnhealth;
```

Comment:
I've tested this and it seems to work properly, did I miss anything?

In case this is later used to replace the rest of color-by-health code, there's two things I'd like to mention: we can also account for the Invulnerability case; and if `mobj->player`, maybe we could use `maxhealth` instead of `mobj->info->spawnhealth` to adapt the thresholds to max health values different than 100.

Reply:
I'd rather like this function to take two arguments: `static char* ColorByHealth(int health, int maxhealth)`. Then it would be called as `ColorByHealth(plr->health, 100)` from the color-by-player-health code path and as `ColorByHealth(crosshair_target->health, crosshair_target->info->spawnhealth) from the color-by-target-health code path. Inside the function, the health value passed as the first argument would have to get normalized by the maxhealth value: `health = 100 * health / maxhealth`.